workers/unity/Packages/com.improbable.gdk.core/Utility/EntityBuilder.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Worker.Core;
+
+namespace Improbable.Gdk.Core
+{
+    public class EntityBuilder
+    {
+        private readonly Entity entity;
+        private Acl acl = new Acl();
+        private bool hasBuiltOnce;
+
+        private readonly HashSet<uint> componentsAdded = new HashSet<uint>();
+
+        public static EntityBuilder Begin()
+        {
+            return new EntityBuilder();
+        }
+
+        protected EntityBuilder()
```

Comment:
Why protected instead of private? Are we expecting this to be inherited?

Reply:
This was copied-pasted from the old one - but yes I agree its probably best to be private.